src/glmfit.jl

```diff
@@ -68,25 +68,21 @@ function updatemu!{T<:FPVector}(r::GlmResp{T}, linPr::T)
 end
 
 function wrkresp(r::GlmResp)
-    if length(r.offset) > 0
-        tmp = r.eta - r.offset
-        broadcast!(+, tmp, tmp, r.wrkresid)
-    else
-        r.eta + r.wrkresid
-    end
+    tmp = r.eta + r.wrkresid
+    isempty(r.offset) ? tmp : broadcast!(-, tmp, tmp, r.offset)
 end
 
 function wrkwt!(r::GlmResp)
     wrkwts = r.wrkwts
     mueta = r.mueta
     var = r.var
-    if length(r.wts) == 0
-        @simd for i = 1:length(r.var)
+    if isempty(r.wts)
+        @simd for i = eachindex(r.var)
```

Comment:
I think it would be even more correct to write `eachindex(r.var, mueta, var)` (unless we are absolutely sure these arrays will always have the same type and structure). Same below.